resources/codegen/tests/generators/test_python.py

```diff
@@ -0,0 +1,681 @@
+        self, local_and_external_trait_specification
+    ):
+        a_traits_data = TraitsData()
+        a_specification = local_and_external_trait_specification(a_traits_data)
+        a_trait = a_specification.openassetioCodegenTestAllNamespaceNoPropertiesTrait()
+        assert a_trait._data is a_traits_data  # pylint: disable=protected-access
+
+
+class Test_generate:
+    def test_when_files_created_then_creation_callback_is_called(
+        self, declaration_exotic_values, creations_exotic_values, tmp_path_factory
+    ):
+        output_dir = tmp_path_factory.mktemp("test_python_generate_callback")
+        expected = [os.path.join(output_dir, p) for p in creations_exotic_values]
+
+        actual = []
+
+        def creation_callback(path):
+            actual.append(path)
+
+        python_generator.generate(
+            declaration_exotic_values,
+            {},
+            output_dir,
+            creation_callback,
+            logging.Logger("Test_generate"),
+        )
+
+        assert actual == expected
+
+    def test_when_names_invalid_then_warnings_are_logged(
+        self,
+        declaration_exotic_values,
+        warnings_exotic_values,
+        a_capturing_logger,
+        tmp_path_factory,
+    ):
+        output_dir = tmp_path_factory.mktemp("test_python_generate_warnings")
+        python_generator.generate(
+            declaration_exotic_values, {}, output_dir, lambda _: _, a_capturing_logger
+        )
+
+        assert a_capturing_logger.handlers[0].messages == warnings_exotic_values
+
+
+#
+# Fixtures
+#
+
+
+@pytest.fixture(scope="module")
+def generated_path(
+    yaml_path_all, yaml_path_traits_only, yaml_path_specifications_only, tmp_path_factory
+):
+    output_dir = tmp_path_factory.mktemp("generated_path")
+
+    def creation_callback(_):
+        pass
+
+    # As there are dependencies between the different packages, we need
+    # to generate them all together to avoid import errors.
+    for description in (yaml_path_all, yaml_path_traits_only, yaml_path_specifications_only):
+        generate(
+            description_path=description,
+            output_directory=output_dir,
+            languages=[
+                "python",
+            ],
+            creation_callback=creation_callback,
+            logger=logging.Logger(name="Capturing logger"),
+        )
+
+    return output_dir
+
+
+@pytest.fixture
+def extended_python_path(generated_path):
+    """
+    Temporarily extends sys.path to include the generated code directory.
+    """
+    packages_path = os.path.join(generated_path, "python")
+    sys.path.insert(0, packages_path)
+    yield packages_path
+    sys.path.pop(0)
+
+
+@pytest.fixture
+def module_all(extended_python_path):
+    """
+    Retrieves the python module corresponding to the 'all' description.
+    """
+    # pylint: disable=import-error,unused-argument,import-outside-toplevel
+    import openassetio_codegen_test_all
```

Comment:
[consider][minor] Do you think it's worth unloading the module once done with it?  Or perhaps make these fixtures "session" scope so its clear that once imported it stays imported.

Reply:
How about [this](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/3f33fd713801793bb97120e76c81968ba603f9dd) for simplicity?

Reply:
Del'd [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/a8736d7d780921d35c2e19ade3372922a301493a) too.